source/common/config/new_grpc_mux_impl.cc

```diff
@@ -153,13 +155,36 @@ GrpcMuxWatchPtr NewGrpcMuxImpl::addWatch(const std::string& type_url,
 // subscription will enqueue and attempt to send an appropriate discovery request.
 void NewGrpcMuxImpl::updateWatch(const std::string& type_url, Watch* watch,
                                  const std::set<std::string>& resources,
-                                 bool creating_namespace_watch) {
+                                 const bool creating_namespace_watch) {
   ASSERT(watch != nullptr);
   auto sub = subscriptions_.find(type_url);
   RELEASE_ASSERT(sub != subscriptions_.end(),
                  fmt::format("Watch of {} has no subscription to update.", type_url));
-  auto added_removed = sub->second->watch_map_.updateWatchInterest(watch, resources);
-  if (creating_namespace_watch) {
+  // If this is a glob collection subscription, we need to compute actual context parameters.
+  std::set<std::string> xdstp_resources;
+  // TODO(htuch): add support for resources beyond glob collections, the constraints below around
+  // resource size and ID reflect the progress of the xdstp:// implementation.
+  if (!resources.empty() && XdsResourceIdentifier::hasXdsTpScheme(*resources.begin())) {
+    // Callers must be asking for a single resource, the collection.
+    ASSERT(resources.size() == 1);
```

Comment:
@asraa @htuch 

Should this be a RELEASE_ASSERT?

Reply:
No, this is an internal invariant, with core control plane code, so this is just documenting a contract that is known to be true.